openpype/hosts/unreal/plugins/load/load_yeticache.py

```diff
@@ -0,0 +1,179 @@
+    icon = "pagelines"
+    color = "orange"
+
+    @staticmethod
+    def get_task(filename, asset_dir, asset_name, replace):
+        task = unreal.AssetImportTask()
+        options = unreal.AbcImportSettings()
+
+        task.set_editor_property('filename', filename)
+        task.set_editor_property('destination_path', asset_dir)
+        task.set_editor_property('destination_name', asset_name)
+        task.set_editor_property('replace_existing', replace)
+        task.set_editor_property('automated', True)
+        task.set_editor_property('save', True)
+
+        task.options = options
+
+        return task
+
+    @staticmethod
+    def is_groom_module_active():
+        """
+        Check if Groom plugin is active.
+
+        This is a workaround, because the Unreal python API don't have
+        any method to check if plugin is active.
+        """
+        prj_file = unreal.Paths.get_project_file_path()
+
+        with open(prj_file, "r") as fp:
+            data = json.load(fp)
+
+        plugins = data.get("Plugins")
+
+        if not plugins:
+            return False
+
+        plugin_names = [p.get("Name") for p in plugins]
+
+        return "HairStrands" in plugin_names
+
+    def load(self, context, name, namespace, options):
+        """Load and containerise representation into Content Browser.
+
+        This is two step process. First, import FBX to temporary path and
+        then call `containerise()` on it - this moves all content to new
+        directory and then it will create AssetContainer there and imprint it
+        with metadata. This will mark this path as container.
+
+        Args:
+            context (dict): application context
+            name (str): subset name
+            namespace (str): in Unreal this is basically path to container.
+                             This is not passed here, so namespace is set
+                             by `containerise()` because only then we know
+                             real path.
+            data (dict): Those would be data to be imprinted. This is not used
+                         now, data are imprinted by `containerise()`.
+
+        Returns:
+            list(str): list of container content
+
+        """
+        # Check if Groom plugin is active
+        if not self.is_groom_module_active():
+            raise RuntimeError("Groom plugin is not activated.")
+
+        # Create directory for asset and Ayon container
+        root = "/Game/Ayon/Assets"
+        asset = context.get('asset').get('name')
+        suffix = "_CON"
+        asset_name = f"{asset}_{name}" if asset else f"{name}"
+
+        tools = unreal.AssetToolsHelpers().get_asset_tools()
+        asset_dir, container_name = tools.create_unique_asset_name(
+            f"{root}/{asset}/{name}", suffix="")
+
+        unique_number = 1
+        while unreal.EditorAssetLibrary.does_directory_exist(
+            f"{asset_dir}_{unique_number:02}"
+        ):
+            unique_number += 1
```

Comment:
I don't get what this `while` is trying to do? Would it raise if the dir doesnt exist? We check again in line `107`?

Reply:
We load the asset in a unique folder, so if the asset already exists (`asset_01`), it would be loaded as `asset_02`. The while exists so that if we have deleted one asset and i.e. we have `asset_01` and `asset_03`, `02` would free again, so we would load it as `asset_02`.